lib/relation-definition.js

```diff
@@ -1356,10 +1394,10 @@ BelongsTo.prototype.related = function (refresh, params) {
           cb(err);
         }
       });
-      return modelInstance[fk];
+      return cb.promise || modelInstance[fk];
```

Comment:
Same comment as above - `related()` should not support promise API, thus `cb.promise` is never set and this change can be reverted.